### ISSUE TITLE
cmd/cue: nested module is not excluded from consideration in parent module

### ISSUE THREAD

**Issue by rogpeppe** [cue-lang/cue#2707]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version v0.0.0-20231113145402-9dd0f82df8e8

go version devel go1.22-3188758653 Fri Nov 10 21:25:30 2023 +0000
      -buildmode exe
       -compiler gc
  DefaultGODEBUG httplaxcontentlength=1,httpmuxgo121=1,panicnil=1
     CGO_ENABLED 1
          GOARCH amd64
            GOOS linux
         GOAMD64 v1
             vcs git
    vcs.revision 9dd0f82df8e8608920b4044c40d977115a5cc95b
        vcs.time 2023-11-13T14:54:02Z
    vcs.modified false
</pre>

### Does this issue reproduce with the latest stable release?

Yes

### What did you do?

```
! exec cue export example.com/x/y
-- cue.mod/module.cue --
module: "example.com/x"

-- x.cue --
package x

a: 1

-- y/cue.mod/module.cue --
module: "other.com/y"

-- y/y.cue --
package y

a: 2
```

### What did you expect to see?

A passing test.

### What did you see instead?

```
> ! exec cue export example.com/x
[stdout]
{
    "a": 2
}
FAIL: /tmp/testscript4064672087/x.txtar/script.txtar:1: unexpected command success
```

The `cue.mod/module.cue` in the subdirectory should probably be respected and the package excluded from consideration as part of the module that it's inside.

**Comment by mvdan** [comment]
Is your testscript perhaps incomplete? The command that you show doesn't relate to the module `other.com/y` or the package `y` at all. I don't see how `cue export example.com/x` should fail. I could see `cue export ./y` should fail, for example, but that's not what you've got in the testscript.

**Comment by rogpeppe** [comment]
> Is your testscript perhaps incomplete? The command that you show doesn't relate to the module `other.com/y` or the package `y` at all. I don't see how `cue export example.com/x` should fail. I could see `cue export ./y` should fail, for example, but that's not what you've got in the testscript.

Yup, good call thanks: I did indeed get the repro wrong. The testscript is now fixed and the issue is still as described in the subject.
